gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/CodeableConceptMapperTest.java

```diff
@@ -468,8 +470,202 @@ public void mapSnomedCodeInTranslationWithDisplayNameWithOriginalText() {
         assertThat(codeableConcept.getCodingFirstRep().getDisplay()).isEqualTo("blood pressure reading");
     }
 
+    @Test
+    public void mapKnownNonSnomedCodeInCodeWithOriginalText() {
+        var inputXML = """
+                <?xml version="1.0" encoding="UTF-8"?>
+                <code xmlns="urn:hl7-org:v3" code="ALLERGY138185NEMIS" codeSystem="2.16.840.1.113883.2.1.6.3"
+                displayName="Adverse reaction to Comirnaty Covid-19 mRna Vaccine">
+                    <originalText>Test original text</originalText>
+                </code>
+                """;
+
+        var codedData = unmarshallCodeElementFromXMLString(inputXML);
+        var codeableConcept = codeableConceptMapper.mapToCodeableConcept(codedData);
+
+        assertThat(codeableConcept.getText()).isEqualTo("Test original text");
```

Comment:
I feel like these asserts might be clearer if they were using the same pattern as the XML in the setup, lines (475-481). That way you could clearly see both the XML and the JSON it would convert to.

Reply:
Will address in a later PR